lib/web/apiserver_test.go

```diff
@@ -1476,63 +1476,45 @@ func TestActiveSessions(t *testing.T) {
-	require.NoError(t, err)
-
-	// Create a session with empty ClusterName.
-	sess1 := session.Session{
-		ClusterName:    "",
-		ServerID:       string(session.NewID()),
-		ID:             session.NewID(),
-		Namespace:      apidefaults.Namespace,
-		Login:          "foo",
-		Created:        time.Date(2009, time.November, 10, 23, 0, 0, 0, time.UTC),
-		LastActive:     time.Date(2009, time.November, 10, 23, 0, 0, 0, time.UTC),
-		TerminalParams: session.TerminalParams{W: 100, H: 100},
-	}
-	err = nodeClient.CreateSession(sess1)
-	require.NoError(t, err)
-
-	// Retrieve the session with the empty ClusterName.
-	pack := s.authPack(t, "baz")
-	res, err := pack.clt.Get(s.ctx, pack.clt.Endpoint("webapi", "sites", s.server.ClusterName(), "sessions", sess1.ID.String()), url.Values{})
-	require.NoError(t, err)
+	sid := session.NewID()
+	pack := s.authPack(t, "foo")
 
-	// Test that empty ClusterName and ServerHostname got set.
-	var sessionResult *session.Session
-	err = json.Unmarshal(res.Bytes(), &sessionResult)
+	ws, err := s.makeTerminal(t, pack, sid)
 	require.NoError(t, err)
-	require.Equal(t, s.server.ClusterName(), sessionResult.ClusterName)
-	require.Equal(t, sess1.ServerID, sessionResult.ServerHostname)
+	t.Cleanup(func() { require.NoError(t, ws.Close()) })
 
-	// Create another session to test sessions list.
-	sess2 := sess1
-	sess2.ID = session.NewID()
-	sess2.ServerID = string(session.NewID())
-	err = nodeClient.CreateSession(sess2)
-	require.NoError(t, err)
+	termHandler := newTerminalHandler()
+	stream := termHandler.asTerminalStream(ws)
 
-	// Retrieve sessions list.
-	res, err = pack.clt.Get(s.ctx, pack.clt.Endpoint("webapi", "sites", s.server.ClusterName(), "sessions"), url.Values{})
+	// To make sure we have a session.
+	_, err = io.WriteString(stream, "echo vinsong\r\n")
 	require.NoError(t, err)
 
-	var sessionList *siteSessionsGetResponse
-	err = json.Unmarshal(res.Bytes(), &sessionList)
+	// Make sure server has replied.
+	err = waitForOutput(stream, "vinsong")
 	require.NoError(t, err)
 
-	s1 := sessionList.Sessions[0]
-	s2 := sessionList.Sessions[1]
-
-	require.Equal(t, s.server.ClusterName(), s1.ClusterName)
-	require.Equal(t, s.server.ClusterName(), s2.ClusterName)
-	require.Equal(t, s1.ServerID, s1.ServerHostname)
-	require.Equal(t, s2.ServerID, s2.ServerHostname)
+	// Make sure this tracker appears in the list of active trackers.
+	var sessResp *siteTrackersGetResponse
+	require.Eventually(t, func() bool {
+		re, err := pack.clt.Get(s.ctx, pack.clt.Endpoint("webapi", "sites", s.server.ClusterName(), "trackers"), url.Values{})
+		require.NoError(t, err)
+		require.NoError(t, json.Unmarshal(re.Bytes(), &sessResp))
+		return len(sessResp.Trackers) == 1
+	}, time.Second*15, time.Millisecond*250)
+
+	require.Len(t, sessResp.Trackers, 1)
+	tracker := sessResp.Trackers[0]
+	require.Equal(t, s.node.GetNamespace(), tracker.GetMetadata().Namespace)
+	require.Greater(t, len(tracker.Spec.Participants), 0)
```

Comment:
Is there a `require.NotEmpty` we can use here?

Reply:
Yep